src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationContext.cs

```diff
@@ -73,5 +76,30 @@ public interface IDurableOrchestrationContext : IInterleavingContext
         /// </remarks>
         /// <param name="customStatusObject">The JSON-serializeable value to use as the orchestrator function's custom status.</param>
         void SetCustomStatus(object customStatusObject);
+
+        /// <summary>
+        /// Makes an HTTP call to the specified uri.
+        /// </summary>
+        /// <param name="method">HttpMethod used for api call.</param>
+        /// <param name="uri">uri used to make the HTTP call.</param>
+        /// <returns>A <see cref="Task{DurableHttpResponse}"/>Result of the HTTP call.</returns>
+        Task<DurableHttpResponse> CallHttpAsync(HttpMethod method, Uri uri);
+
+        /// <summary>
+        /// Makes an HTTP call to the specified uri with authentication information
+        /// to get a token.
+        /// </summary>
+        /// <param name="method">HttpMethod used for api call.</param>
+        /// <param name="uri">uri used to make the HTTP call.</param>
+        /// <param name="tokenSource">information to get a token from a specified service.</param>
+        /// <returns>A <see cref="Task{TResult}"/> Result of the HTTP call.</returns>
+        Task<DurableHttpResponse> CallHttpAsync(HttpMethod method, Uri uri, ITokenSource tokenSource);
```

Comment:
Generally speaking, the simplified method overloads should cover the most common cases. In my opinion, this particular overload with an `ITokenSource` and no payload will not be a common case. Rather, I would suggest you have a `CallHttpAsync` method that takes a method, URL, and optional _request content_, defined like this:

```csharp
Task<DurableHttpResponse> CallHttpAsync(HttpMethod method, Uri uri, string content = null);
```
And to be honest I think the first overload (HTTP method, and URL) probably isn't needed either since requests without payloads are only common for GET requests. I think if we just had the proposed one (in addition to the `DurableHttpRequest`) then that will give us the smallest number of methods to maintain while still having very good ease-of-use.